R/FilteredData.R

```diff
@@ -138,7 +139,39 @@ FilteredData <- R6::R6Class( # nolint
       private$get_filtered_dataset(dataname)$get_dataset_label()
     },
 
+    #' Set external `teal_slice`
+    #'
+    #' Unlike adding new filter from the column, these filters can be added with some pre-specified
+    #' settings. List of `teal_slices` should be a reactive so one can make this list to be dynamic.
+    #' List is accessible in `ui/srv_active` through `ui/srv_available_filters`.
+    #' @param x (`reactive`)\cr
+    #'  should return `teal_slices`
+    #' @return invisible `NULL`
+    set_available_teal_slices = function(x) {
+      checkmate::assert_class(x, "reactive")
+      private$available_teal_slices <- reactive({
+        # we want to limit the available filters to the ones that are relevant for this FilteredData
+        Filter(function(x) x$dataname %in% self$datanames(), x())
+      })
+      invisible(NULL)
+    },
+
+    #' Activate snapshot manager.
+    #'
+    #' Activate snapshot manager module to allow capturing, saving, and restoring
+    #' snapshots of the global application filter state.
+    #'
+    #' @param x (`logical(1)`) flag
+    #' @return invisible `NULL`
+    activate_snapshot_manager = function(x) {
+      checkmate::assert_flag(x)
+      private$snapshot_manager_active <- x
+      invisible(NULL)
+    },
```

Comment:
I think we could expose this as an api attribute, e.g. `module_snapshot`. I know it might be tricky as:
- `teal::teal_slices(module_snapshot)` would call `teal::snapshot_manager_ui/srv` only when `module_specific = TRUE`
- `teal.slice::teal_slices(module_snapshot)` would call `FilteredData::snapshot_manager` (through `ui/srv_active` module).

We can discuss this soon